### PR TITLE
chore: clean deployment

### PR DESCRIPTION
**Pull request by jfrery**
closes https://github.com/zama-ai/concrete-ml-internal/issues/4411
closes https://github.com/zama-ai/concrete-ml-internal/issues/4388
closes https://github.com/zama-ai/concrete-ml-internal/issues/3927
closes https://github.com/zama-ai/concrete-ml-internal/issues/3561
closes https://github.com/zama-ai/concrete-ml-internal/issues/3598
closes https://github.com/zama-ai/concrete-ml-internal/issues/4064
closes https://github.com/zama-ai/concrete-ml-internal/issues/4193

### CODE REVIEW

File: use_case_examples/deployment/server/deploy_to_aws.py [hunk 1]
```diff
@@ -23,7 +23,7 @@
 
 import boto3
 
-from ..deployment.utils import filter_logs, wait_for_connection_to_be_available
+from .utils_server import filter_logs, wait_for_connection_to_be_available
```
Comment (andrei-stoian-zama) [inline use_case_examples/deployment/server/deploy_to_aws.py:26]: are you missing some files here? in the `utils_server` dir?

File: use_case_examples/deployment/server/server_requirements.txt
Comment (andrei-stoian-zama) [inline use_case_examples/deployment/server/server_requirements.txt:1]: don't you need to add the dependencies removed from pyproject.toml here ?

File: src/concrete/ml/sklearn/glm.py [hunk 2]
```diff
@@ -325,7 +325,6 @@ def dump_dict(self) -> Dict:
         # Scikit-Learn
         metadata["alpha"] = self.alpha
         metadata["fit_intercept"] = self.fit_intercept
-        metadata["solver"] = self.solver
```
Comment (andrei-stoian-zama) [inline src/concrete/ml/sklearn/glm.py:328]: any idea why these are removed ?

File: src/concrete/ml/sklearn/base.py [hunk 3]
```diff
@@ -1291,7 +1292,8 @@ class BaseTreeEstimatorMixin(BaseEstimator, sklearn.base.BaseEstimator, ABC):
     #: Model's base framework used, either 'xgboost' or 'sklearn'. Is set for each subclasses.
     framework: str
 
-    def __init_subclass__(cls):
+    def __init_subclass__(cls, *args, **kwargs):
+        super().__init_subclass__(*args, **kwargs)
```
Comment (andrei-stoian-zama) [inline src/concrete/ml/sklearn/base.py:1296]: can you explain these changes please ?

File: docs/guides/client_server.md [hunk 4]
```diff
@@ -32,19 +32,3 @@ The server-side implementation of a Concrete ML model follows the diagram above.
 ## Example notebook
 
 For a complete example, see [the client-server notebook](../advanced_examples/ClientServer.ipynb) or [the use-case examples](../../use_case_examples/deployment/).
-
```
Comment (andrei-stoian-zama) [inline docs/guides/client_server.md:35]: maybe move these lines to the `deployment` directory in `use_case_examples` ?

**Review by andrei-stoian-zama — CHANGES_REQUESTED** [review]
Great! I find that removing dependencies is very satisfying :)

**Review by fd0r — APPROVED** [review]
Some comments from @andrei-stoian-zama but otherwise fine with me

File: use_case_examples/deployment/server/deploy_to_aws.py (hunk 1 above)
Comment (jfrery) [reply use_case_examples/deployment/server/deploy_to_aws.py:26]: utils_server is a py file. Should be good.

File: src/concrete/ml/sklearn/glm.py (hunk 2 above)
Comment (jfrery) [reply src/concrete/ml/sklearn/glm.py:328]: This comes from the new test we added recently 

https://github.com/zama-ai/concrete-ml/blob/6921c91c6e44e89b662236501f8b5f10bfeeff4a/tests/sklearn/test_sklearn_models.py#L2057

It highlighted a few discrepancies with sklearn models. Here `solver` isn't actually part of this model in sklearn.

File: src/concrete/ml/sklearn/base.py (hunk 3 above)
Comment (jfrery) [reply src/concrete/ml/sklearn/base.py:1296]: Still related to the new test.

**Review by andrei-stoian-zama — CHANGES_REQUESTED** [review]
your new commit seems to delete all the deployment use case

**Review by andrei-stoian-zama — DISMISSED** [review]
Looks good! How can we test the cifar/sentiment/breast cancer deployment use cases using the CI ?

**Review by andrei-stoian-zama — APPROVED** [review]

**Pull request merged**